vision.py

```diff
@@ -45,25 +45,35 @@ def process_image(self, frame: np.ndarray) -> Optional[Tuple[float, float]]:
         filtered_contours = self.filter_contours(contours)
         if len(filtered_contours) == 0:
             return None
+        self.contourAreas = [cv2.contourArea(c) for c in filtered_contours]
         groups = self.group_contours(filtered_contours)
         if len(groups) == 0:
             return None
-        best_group = self.rank_groups(contours, groups)
-        return self.get_values(contours, best_group, (frame.shape[1], frame.shape[0]))
+        best_group = self.rank_groups(filtered_contours, groups)
+        if best_group is None:
+            return None
+        return self.get_values(filtered_contours, best_group, (frame.shape[1], frame.shape[0]))
 
     def preprocess(self, frame: np.ndarray) -> np.ndarray:
         """Creates a mask of expected target green color
         """
         hsv = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
-        mask = cv2.inRange(hsv, TARGET_HSV_LOW, TARGET_HSV_HIGH)
-        mask = cv2.erode(mask, ERODE_DILATE_KERNEL)
-        mask = cv2.dilate(mask, ERODE_DILATE_KERNEL)
-        return mask
+        self.mask = cv2.inRange(hsv, TARGET_HSV_LOW, TARGET_HSV_HIGH)
```

Comment:
Should we pass in `dst` so that it can be reused? If no, any reason to make it an attribute?